src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -0,0 +1,714 @@
+                                 ares_resolver, error_msg.c_str());
+    ares_resolver->event_engine_->Run(
+        [callback = std::move(callback),
+         status = AresStatusToAbslStatus(status, error_msg)]() mutable {
+          callback(status);
+        });
+    return;
+  }
+  std::vector<EventEngine::ResolvedAddress> result;
+  for (size_t i = 0; hostent->h_addr_list[i] != nullptr; i++) {
+    switch (hostent->h_addrtype) {
+      case AF_INET6: {
+        size_t addr_len = sizeof(struct sockaddr_in6);
+        struct sockaddr_in6 addr;
+        memset(&addr, 0, addr_len);
+        memcpy(&addr.sin6_addr, hostent->h_addr_list[i],
+               sizeof(struct in6_addr));
+        addr.sin6_family = static_cast<unsigned char>(hostent->h_addrtype);
+        addr.sin6_port = htons(hostname_qa->port);
+        result.emplace_back(reinterpret_cast<const sockaddr*>(&addr), addr_len);
+        char output[INET6_ADDRSTRLEN];
+        ares_inet_ntop(AF_INET6, &addr.sin6_addr, output, INET6_ADDRSTRLEN);
+        GRPC_ARES_RESOLVER_TRACE_LOG(
+            "request:%p c-ares resolver gets a AF_INET6 result: \n"
+            "  addr: %s\n  port: %d\n  sin6_scope_id: %d\n",
+            ares_resolver, output, hostname_qa->port, addr.sin6_scope_id);
+        break;
+      }
+      case AF_INET: {
+        size_t addr_len = sizeof(struct sockaddr_in);
+        struct sockaddr_in addr;
+        memset(&addr, 0, addr_len);
+        memcpy(&addr.sin_addr, hostent->h_addr_list[i], sizeof(struct in_addr));
+        addr.sin_family = static_cast<unsigned char>(hostent->h_addrtype);
+        addr.sin_port = htons(hostname_qa->port);
+        result.emplace_back(reinterpret_cast<const sockaddr*>(&addr), addr_len);
+        char output[INET_ADDRSTRLEN];
+        ares_inet_ntop(AF_INET, &addr.sin_addr, output, INET_ADDRSTRLEN);
+        GRPC_ARES_RESOLVER_TRACE_LOG(
+            "request:%p c-ares resolver gets a AF_INET result: \n"
+            "  addr: %s\n  port: %d\n",
+            ares_resolver, output, hostname_qa->port);
+        break;
+      }
+    }
+  }
+  ares_resolver->event_engine_->Run(
+      [callback = std::move(callback), result = std::move(result)]() mutable {
+        callback(std::move(result));
+      });
+}
+
+void AresResolver::OnSRVQueryDoneLocked(void* arg, int status, int /*timeouts*/,
+                                        unsigned char* abuf, int alen) {
+  std::unique_ptr<QueryArg> qa(static_cast<QueryArg*>(arg));
+  auto* ares_resolver = qa->ares_resolver;
+  auto nh = ares_resolver->callback_map_.extract(qa->callback_map_id);
+  GPR_ASSERT(!nh.empty());
+  GPR_ASSERT(
+      absl::holds_alternative<EventEngine::DNSResolver::LookupSRVCallback>(
+          nh.mapped()));
+  auto callback = absl::get<EventEngine::DNSResolver::LookupSRVCallback>(
+      std::move(nh.mapped()));
+  if (status != ARES_SUCCESS) {
+    std::string error_msg = absl::StrFormat(
+        "c-ares status is not ARES_SUCCESS qtype=SRV name=%s: %s",
```

Comment:
Suggest the string "SRV lookup failed for %s: %s".

Reply:
Done.